_posts/2020-09-14-Introduction-to-API-Management.md

```diff
@@ -0,0 +1,292 @@
+If the Gateway is installed in the same organizational structure, it’s not required to have your API’s implementing additional cross cutting functionalities.
+The API gateway will be responsible of performing the necessary checks before allowing the clients to access the Data.
+I do hear you thinking already “We’re not going to install an additional tool, only for splitting out security rules from the API’s", but that’s not the only thing an API management tool can do.
+
+### Responsibilities of an API Management tool
+I listed out the most important responsibilities for you, so that you can understand the power and the actual benefits of an API Management tool:            
+     
+1.	Authentication 
+    * Enables one of the most important parts of security on an API. Validating the person’s identity!  
+    There are different kinds of Authentication:   
+        * Http Basic Authentication
+            * Using this approach, a user agent simply provides a username and password to prove their authentication.  
+            This approach does not require cookies, session ID’s, or login pages because it leverages the HTTP header itself. While simple to use, this method of authentication is vulnerable to attacks that could capture the user’s credentials in transit.
+            
+        * OAuth
+            * OAuth 2.0 is the best choice for identifying personal user accounts and granting proper permissions.  
+            With this method, the user logs into a system. That system will then request authentication, usually in the form of a token. The user will then forward this request to an authentication server, which will either reject or allow this authentication.
+        
+        * OpenID Connect & OAuth 2.0 API
+            * OpenID Connect is a simple identity layer on top of the OAuth 2.0 protocol, which allows computing clients to verify the identity of an end-user based on the authentication performed by an authorization server, as well as to obtain basic profile information about the end-user in an interoperable and REST-like manner.
+        
+        * API Keys
+            * An API key is an identifier meant to identify the origin of web service requests (or similar types of requests).  
+            A key is generated the first time a user attempts to gain authorized access to a system through registration. From there, the API key becomes associated with a secret token, and is submitted alongside requests going forward. When the user attempts to re-enter the system, their unique key is used to prove that they are the same user as before. This API Authentication Method is very fast and reliable but is frequently misused. More importantly, this method of authentication is not a method of authorization.   
+2.	Authorization 
+    * Enables fine grained authorization to API resources based on authenticated user roles.  
+    This involves checking resources that the user is authorized to access or modify via defined roles.

```

Comment:
The API gateway just verifies the token with an external authorization service and propagates it to the downstream service. Here the service will enable fine-grained authorization to API resources.

Reply:
I changed it